### PR TITLE
[Doc] update case expr

### PR DESCRIPTION
**Pull request by evelynzhaojie**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: docs/sql-reference/sql-functions/condition-functions/case_when.md [hunk 1]
```diff
@@ -35,37 +46,102 @@ CASE WHEN expression = expression1 THEN result1
...
-`expressionN` are compatible in data type.
+- `conditionN`: the condition that can evaluate to a BOOLEAN value.
 
-`resultN` are compatible in data type.
+- `resultN` must be compatible in data types.
 
 ## Return value
 
-The return value is of the common type of all types from `then-clause`.
+The return value is of the same type as the result in the THEN clause.
```
Comment (fzhedu) [inline docs/sql-reference/sql-functions/condition-functions/case_when.md:59]: the common type from then-clause

File: docs/sql-reference/sql-functions/condition-functions/case_when.md [hunk 2]
```diff
@@ -35,37 +46,102 @@ CASE WHEN expression = expression1 THEN result1
...
-| gender  | gender_str  |
-+---------+-------------+
-| 1       | male        |
-| 0       | female      |
-| -1      | NULL        |
-+---------+-------------+
+CREATE TABLE test_case(
+    name          STRING,
+    gender         INT,
+    score          FLOAT
```
Comment (fzhedu) [inline docs/sql-reference/sql-functions/condition-functions/case_when.md:69]: no nedd score

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**